### PR TITLE
Feature/dump csv

### PR DESCRIPTION
**Pull request by kimhoang73**
On the apps detail page, there's now a button you can click to download all the engagement data for the app in a CSV file

### CODE REVIEW

**Review by kyriakipapadimitriou — APPROVED** [review]
Precise definition of the export method and testing! Good job

**Review by julies642 — APPROVED** [review]
Nice testing for the csv file. Looks good to me!

File: app/controllers/engagements_controller.rb [hunk 1]
```diff
@@ -45,6 +48,28 @@ def destroy
...
+    eng_hash["iterations"] = iterations_hash
+    # make entire thing into a json
+    eng_params = JSON.parse(eng_hash.to_json)
+    keys = eng_params.keys
+    values = eng_params.values
+    file = CSV.generate do |csv|
+      csv << keys
+      csv << values
+    end
+    send_data(file, :filename => "engagement.csv")
```
Comment (csungwon) [inline app/controllers/engagements_controller.rb:70]: Might be a good idea to have different file names for each engagement. e.g. file name by engagement id. But push to next iteration!

**Review by csungwon — APPROVED** [review]
Everything looks good! Nice work!

File: app/controllers/engagements_controller.rb (hunk 1 above)
Comment (kimhoang73) [reply app/controllers/engagements_controller.rb:70]: yes that's a good idea!

**Pull request merged**